spark/sql/dataframe.go

```diff
@@ -148,6 +148,8 @@ type DataFrame interface {
 	Rollup(ctx context.Context, cols ...column.Convertible) *GroupedData
 	// SameSemantics returns true if the other DataFrame has the same semantics.
 	SameSemantics(ctx context.Context, other DataFrame) (bool, error)
+	// Sample samples a data frame
+	Sample(ctx context.Context, withReplacement *bool, fraction float64, seed *int64) (DataFrame, error)
```

Comment:
I believe a more Golang idiomatic approach would be to have multiple methods here

```
Sample(ctx) (DataFrame, error)
SampleWithFraction(ctx, fraction) (DataFrame, error)
SampleWithReplacementFractionSeed(ctx, withReplacement, fraction, seed) (DataFrame, error)
```

Reply:
Done.